zzk/service/hoststate.go

```diff
@@ -119,192 +151,264 @@ func (l *HostStateListener) Spawn(shutdown <-chan interface{}, stateID string) {
 		InstanceID: instanceID,
 	}
 
-	var containerExit <-chan time.Time
-	defer func() {
-
-		// stop the container
-		if err := l.handler.StopContainer(serviceID, instanceID); err != nil {
-			logger.WithError(err).Error("Could not stop container")
-		} else if containerExit != nil {
-			// wait for the container to exit
-			time := <-containerExit
-			logger.WithField("terminated", time).Debug("Container exited")
-		}
+	// reattach to orphaned container
+	ssdat, containerExit := l.getExistingState(stateID)
+	// TODO: Should we go ahead and pull the service state node from zk and make sure it matches?
```

Comment:
No, because this is the only place the service state node will ever be modified.